components/lib/datatable/BodyCell.js

```diff
@@ -189,6 +189,8 @@ export const BodyCell = React.memo((props) => {
                 closeCell(event);
             } else {
                 event.preventDefault();
+
+                return newRowData;
```

Comment:
@melloware I don't think this is right, but take my comment with a grain of salt, I did not spent too much time with this code. I think you should return newRowData regardless of the validator return true or false, otherwise you set editingRowDataState to undefined in the valid case and nothing will set it to the correct value next time to switch back to editor mode.

Reply:
Ok i thought so your comment confused me. I will fix.